docs/src/dist-training-torch/iterable_style.md

```diff
@@ -0,0 +1,90 @@
+# Training on Stream data with `LanceDataset` (Iterable-Style)
+
+The basic iterable-style pattern is for use cases where you need to stream data sequentially or implement custom sampling logic. This can be due to various reasons like - your data is tool large to fit in a node or in a massively parallel multi-node multi-GPU setting, you need to read data from a central source, etc. It uses the `LanceDataset` and a Lance-native sampler.
```

Comment:
> your data is tool large to fit in a node or in a massively parallel multi-node multi-GPU setting, you need to read data from a central source

I don't understand these reasons.  A map dataset does not materialize the entire dataset in memory does it?

Reply:
Yeah let me rephrase this part to be lance specific